emission/analysis/classification/inference/labels/pipeline.py

```diff
@@ -0,0 +1,121 @@
+# Standard imports
+import logging
+
+# Our imports
+import emission.storage.pipeline_queries as epq
+import emission.storage.timeseries.abstract_timeseries as esta
+import emission.storage.decorations.analysis_timeseries_queries as esda
+import emission.core.wrapper.labelprediction as ecwl
+
+def infer_labels(user_id):
+    time_query = epq.get_time_range_for_label_inference(user_id)
+    try:
+        lip = LabelInferencePipeline()
+        lip.user_id = user_id
+        lip.run_prediction_pipeline(user_id, time_query)
+        if lip.last_trip_done is None:
+            logging.debug("After run, last_trip_done == None, must be early return")
+        epq.mark_label_inference_done(user_id, lip.last_trip_done)
+    except:
+        logging.exception("Error while inferring labels, timestamp is unchanged")
+        epq.mark_label_inference_failed(user_id)
+
+# Code structure based on emission.analysis.classification.inference.mode.pipeline
+# and emission.analysis.classification.inference.mode.rule_engine
+class LabelInferencePipeline:
+    def __init__(self):
+        self._last_trip_done = None
+    
+    @property
+    def last_trip_done(self):
+        return self._last_trip_done
+
+    def run_prediction_pipeline(self, user_id, time_range):
+        self.ts = esta.TimeSeries.get_time_series(user_id)
+        self.toPredictTrips = esda.get_entries(
+            esda.CLEANED_TRIP_KEY, user_id, time_query=time_range)
+        for trip in self.toPredictTrips:
+            prediction = predict_trip(trip)
+            lp = ecwl.Labelprediction()
+            lp.trip_id = trip.get_id()
+            lp.prediction = prediction
+            lp.start_ts = trip.data.start_ts
+            lp.end_ts = trip.data.end_ts
+            # Insert the Labelprediction into the database as its own independent document
+            self.ts.insert_data(self.user_id, esda.INFERRED_LABELS_KEY, lp)
```

Comment:
Am I understanding correctly that for mode inference, `inference/prediction` is to store each algorithm's result and `analysis/inferred_section` is to store the final (perhaps ensemble) value? If that's the case, then yes, I suppose it might be useful to do the same for label inference.

Reply:
Correct. At least, that is the long-term goal. Right now, there is only one implementation 😄

Reply:
Implemented in https://github.com/e-mission/e-mission-server/pull/825/commits/d0136fc8c041739bf2809f74b3e8b44846957785.